FLIGHT PRICE PREDICTION

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib 
%matplotlib inline

In [3]:
df=pd.read_excel(r'C:\Users\Welcome\Downloads\Data_Train.xlsx') 


In [4]:
df

Airline Date_of_Journey    Source Destination  \
0           IndiGo      24/03/2019  Banglore   New Delhi   
1        Air India       1/05/2019   Kolkata    Banglore   
2      Jet Airways       9/06/2019     Delhi      Cochin   
3           IndiGo      12/05/2019   Kolkata    Banglore   
4           IndiGo      01/03/2019  Banglore   New Delhi   
...            ...             ...       ...         ...   
10678     Air Asia       9/04/2019   Kolkata    Banglore   
10679    Air India      27/04/2019   Kolkata    Banglore   
10680  Jet Airways      27/04/2019  Banglore       Delhi   
10681      Vistara      01/03/2019  Banglore   New Delhi   
10682    Air India       9/05/2019     Delhi      Cochin   

                       Route Dep_Time  Arrival_Time Duration Total_Stops  \
0                  BLR → DEL    22:20  01:10 22 Mar   2h 50m    non-stop   
1      CCU → IXR → BBI → BLR    05:50         13:15   7h 25m     2 stops   
2      DEL → LKO → BOM → COK    09:25  04:25 10 Jun      19h     2 stops   
3            CCU → NAG → BLR    18:05         23:30   5h 25m      1 stop   
4            BLR → NAG → DEL    16:50         21:35   4h 45m      1 stop   
...                      ...      ...           ...      ...         ...   
10678              CCU → BLR    19:55         22:25   2h 30m    non-stop   
10679              CCU → BLR    20:45         23:20   2h 35m    non-stop   
10680              BLR → DEL    08:20         11:20       3h    non-stop   
10681              BLR → DEL    11:30         14:10   2h 40m    non-stop   
10682  DEL → GOI → BOM → COK    10:55         19:15   8h 20m     2 stops   

      Additional_Info  Price  
0             No info   3897  
1             No info   7662  
2             No info  13882  
3             No info   6218  
4             No info  13302  
...               ...    ...  
10678         No info   4107  
10679         No info   4145  
10680         No info   7229  
10681         No info  12648  
10682         No info  11753  

[10683 rows x 11 columns]

In [5]:
df.describe()

Price
count  10683.000000
mean    9087.064121
std     4611.359167
min     1759.000000
25%     5277.000000
50%     8372.000000
75%    12373.000000
max    79512.000000

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [7]:
test=pd.read_excel(r'C:\Users\Welcome\Downloads\Test_set.xlsx')

In [8]:
test

Airline Date_of_Journey    Source Destination  \
0           Jet Airways       6/06/2019     Delhi      Cochin   
1                IndiGo      12/05/2019   Kolkata    Banglore   
2           Jet Airways      21/05/2019     Delhi      Cochin   
3     Multiple carriers      21/05/2019     Delhi      Cochin   
4              Air Asia      24/06/2019  Banglore       Delhi   
...                 ...             ...       ...         ...   
2666          Air India       6/06/2019   Kolkata    Banglore   
2667             IndiGo      27/03/2019   Kolkata    Banglore   
2668        Jet Airways       6/03/2019     Delhi      Cochin   
2669          Air India       6/03/2019     Delhi      Cochin   
2670  Multiple carriers      15/06/2019     Delhi      Cochin   

                Route Dep_Time  Arrival_Time Duration Total_Stops  \
0     DEL → BOM → COK    17:30  04:25 07 Jun  10h 55m      1 stop   
1     CCU → MAA → BLR    06:20         10:20       4h      1 stop   
2     DEL → BOM → COK    19:15  19:00 22 May  23h 45m      1 stop   
3     DEL → BOM → COK    08:00         21:00      13h      1 stop   
4           BLR → DEL    23:55  02:45 25 Jun   2h 50m    non-stop   
...               ...      ...           ...      ...         ...   
2666  CCU → DEL → BLR    20:30  20:25 07 Jun  23h 55m      1 stop   
2667        CCU → BLR    14:20         16:55   2h 35m    non-stop   
2668  DEL → BOM → COK    21:50  04:25 07 Mar   6h 35m      1 stop   
2669  DEL → BOM → COK    04:00         19:15  15h 15m      1 stop   
2670  DEL → BOM → COK    04:55         19:15  14h 20m      1 stop   

                  Additional_Info  
0                         No info  
1                         No info  
2     In-flight meal not included  
3                         No info  
4                         No info  
...                           ...  
2666                      No info  
2667                      No info  
2668                      No info  
2669                      No info  
2670                      No info  

[2671 rows x 10 columns]

In [9]:
final=pd.concat([df,test],ignore_index=True)

In [10]:
final

Airline Date_of_Journey    Source Destination  \
0                 IndiGo      24/03/2019  Banglore   New Delhi   
1              Air India       1/05/2019   Kolkata    Banglore   
2            Jet Airways       9/06/2019     Delhi      Cochin   
3                 IndiGo      12/05/2019   Kolkata    Banglore   
4                 IndiGo      01/03/2019  Banglore   New Delhi   
...                  ...             ...       ...         ...   
13349          Air India       6/06/2019   Kolkata    Banglore   
13350             IndiGo      27/03/2019   Kolkata    Banglore   
13351        Jet Airways       6/03/2019     Delhi      Cochin   
13352          Air India       6/03/2019     Delhi      Cochin   
13353  Multiple carriers      15/06/2019     Delhi      Cochin   

                       Route Dep_Time  Arrival_Time Duration Total_Stops  \
0                  BLR → DEL    22:20  01:10 22 Mar   2h 50m    non-stop   
1      CCU → IXR → BBI → BLR    05:50         13:15   7h 25m     2 stops   
2      DEL → LKO → BOM → COK    09:25  04:25 10 Jun      19h     2 stops   
3            CCU → NAG → BLR    18:05         23:30   5h 25m      1 stop   
4            BLR → NAG → DEL    16:50         21:35   4h 45m      1 stop   
...                      ...      ...           ...      ...         ...   
13349        CCU → DEL → BLR    20:30  20:25 07 Jun  23h 55m      1 stop   
13350              CCU → BLR    14:20         16:55   2h 35m    non-stop   
13351        DEL → BOM → COK    21:50  04:25 07 Mar   6h 35m      1 stop   
13352        DEL → BOM → COK    04:00         19:15  15h 15m      1 stop   
13353        DEL → BOM → COK    04:55         19:15  14h 20m      1 stop   

      Additional_Info    Price  
0             No info   3897.0  
1             No info   7662.0  
2             No info  13882.0  
3             No info   6218.0  
4             No info  13302.0  
...               ...      ...  
13349         No info      NaN  
13350         No info      NaN  
13351         No info      NaN  
13352         No info      NaN  
13353         No info      NaN  

[13354 rows x 11 columns]

In [11]:
final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13354 entries, 0 to 13353
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Date_of_Journey  13354 non-null  object 
 2   Source           13354 non-null  object 
 3   Destination      13354 non-null  object 
 4   Route            13353 non-null  object 
 5   Dep_Time         13354 non-null  object 
 6   Arrival_Time     13354 non-null  object 
 7   Duration         13354 non-null  object 
 8   Total_Stops      13353 non-null  object 
 9   Additional_Info  13354 non-null  object 
 10  Price            10683 non-null  float64
dtypes: float64(1), object(10)
memory usage: 1.1+ MB


In [12]:
final.describe()

Price
count  10683.000000
mean    9087.064121
std     4611.359167
min     1759.000000
25%     5277.000000
50%     8372.000000
75%    12373.000000
max    79512.000000

we should split the values of dates in monthes , date and year

In [13]:
final['Date']=final['Date_of_Journey'].str.split('/').str[0]

In [14]:
final['Month']=final['Date_of_Journey'].str.split('/').str[1]


In [15]:
final['Year']=final['Date_of_Journey'].str.split('/').str[2]

In [16]:
final

Airline Date_of_Journey    Source Destination  \
0                 IndiGo      24/03/2019  Banglore   New Delhi   
1              Air India       1/05/2019   Kolkata    Banglore   
2            Jet Airways       9/06/2019     Delhi      Cochin   
3                 IndiGo      12/05/2019   Kolkata    Banglore   
4                 IndiGo      01/03/2019  Banglore   New Delhi   
...                  ...             ...       ...         ...   
13349          Air India       6/06/2019   Kolkata    Banglore   
13350             IndiGo      27/03/2019   Kolkata    Banglore   
13351        Jet Airways       6/03/2019     Delhi      Cochin   
13352          Air India       6/03/2019     Delhi      Cochin   
13353  Multiple carriers      15/06/2019     Delhi      Cochin   

                       Route Dep_Time  Arrival_Time Duration Total_Stops  \
0                  BLR → DEL    22:20  01:10 22 Mar   2h 50m    non-stop   
1      CCU → IXR → BBI → BLR    05:50         13:15   7h 25m     2 stops   
2      DEL → LKO → BOM → COK    09:25  04:25 10 Jun      19h     2 stops   
3            CCU → NAG → BLR    18:05         23:30   5h 25m      1 stop   
4            BLR → NAG → DEL    16:50         21:35   4h 45m      1 stop   
...                      ...      ...           ...      ...         ...   
13349        CCU → DEL → BLR    20:30  20:25 07 Jun  23h 55m      1 stop   
13350              CCU → BLR    14:20         16:55   2h 35m    non-stop   
13351        DEL → BOM → COK    21:50  04:25 07 Mar   6h 35m      1 stop   
13352        DEL → BOM → COK    04:00         19:15  15h 15m      1 stop   
13353        DEL → BOM → COK    04:55         19:15  14h 20m      1 stop   

      Additional_Info    Price Date Month  Year  
0             No info   3897.0   24    03  2019  
1             No info   7662.0    1    05  2019  
2             No info  13882.0    9    06  2019  
3             No info   6218.0   12    05  2019  
4             No info  13302.0   01    03  2019  
...               ...      ...  ...   ...   ...  
13349         No info      NaN    6    06  2019  
13350         No info      NaN   27    03  2019  
13351         No info      NaN    6    03  2019  
13352         No info      NaN    6    03  2019  
13353         No info      NaN   15    06  2019  

[13354 rows x 14 columns]

In [17]:
final.drop('Date_of_Journey',axis=1,inplace=True)

In [18]:
final

Airline    Source Destination                  Route  \
0                 IndiGo  Banglore   New Delhi              BLR → DEL   
1              Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2            Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK   
3                 IndiGo   Kolkata    Banglore        CCU → NAG → BLR   
4                 IndiGo  Banglore   New Delhi        BLR → NAG → DEL   
...                  ...       ...         ...                    ...   
13349          Air India   Kolkata    Banglore        CCU → DEL → BLR   
13350             IndiGo   Kolkata    Banglore              CCU → BLR   
13351        Jet Airways     Delhi      Cochin        DEL → BOM → COK   
13352          Air India     Delhi      Cochin        DEL → BOM → COK   
13353  Multiple carriers     Delhi      Cochin        DEL → BOM → COK   

      Dep_Time  Arrival_Time Duration Total_Stops Additional_Info    Price  \
0        22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897.0   
1        05:50         13:15   7h 25m     2 stops         No info   7662.0   
2        09:25  04:25 10 Jun      19h     2 stops         No info  13882.0   
3        18:05         23:30   5h 25m      1 stop         No info   6218.0   
4        16:50         21:35   4h 45m      1 stop         No info  13302.0   
...        ...           ...      ...         ...             ...      ...   
13349    20:30  20:25 07 Jun  23h 55m      1 stop         No info      NaN   
13350    14:20         16:55   2h 35m    non-stop         No info      NaN   
13351    21:50  04:25 07 Mar   6h 35m      1 stop         No info      NaN   
13352    04:00         19:15  15h 15m      1 stop         No info      NaN   
13353    04:55         19:15  14h 20m      1 stop         No info      NaN   

      Date Month  Year  
0       24    03  2019  
1        1    05  2019  
2        9    06  2019  
3       12    05  2019  
4       01    03  2019  
...    ...   ...   ...  
13349    6    06  2019  
13350   27    03  2019  
13351    6    03  2019  
13352    6    03  2019  
13353   15    06  2019  

[13354 rows x 13 columns]

In [19]:
final.drop('Additional_Info',axis=1,inplace=True)

In [20]:
final

Airline    Source Destination                  Route  \
0                 IndiGo  Banglore   New Delhi              BLR → DEL   
1              Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2            Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK   
3                 IndiGo   Kolkata    Banglore        CCU → NAG → BLR   
4                 IndiGo  Banglore   New Delhi        BLR → NAG → DEL   
...                  ...       ...         ...                    ...   
13349          Air India   Kolkata    Banglore        CCU → DEL → BLR   
13350             IndiGo   Kolkata    Banglore              CCU → BLR   
13351        Jet Airways     Delhi      Cochin        DEL → BOM → COK   
13352          Air India     Delhi      Cochin        DEL → BOM → COK   
13353  Multiple carriers     Delhi      Cochin        DEL → BOM → COK   

      Dep_Time  Arrival_Time Duration Total_Stops    Price Date Month  Year  
0        22:20  01:10 22 Mar   2h 50m    non-stop   3897.0   24    03  2019  
1        05:50         13:15   7h 25m     2 stops   7662.0    1    05  2019  
2        09:25  04:25 10 Jun      19h     2 stops  13882.0    9    06  2019  
3        18:05         23:30   5h 25m      1 stop   6218.0   12    05  2019  
4        16:50         21:35   4h 45m      1 stop  13302.0   01    03  2019  
...        ...           ...      ...         ...      ...  ...   ...   ...  
13349    20:30  20:25 07 Jun  23h 55m      1 stop      NaN    6    06  2019  
13350    14:20         16:55   2h 35m    non-stop      NaN   27    03  2019  
13351    21:50  04:25 07 Mar   6h 35m      1 stop      NaN    6    03  2019  
13352    04:00         19:15  15h 15m      1 stop      NaN    6    03  2019  
13353    04:55         19:15  14h 20m      1 stop      NaN   15    06  2019  

[13354 rows x 12 columns]

In [21]:
final['Arrival_Time']

0        01:10 22 Mar
1               13:15
2        04:25 10 Jun
3               23:30
4               21:35
             ...     
13349    20:25 07 Jun
13350           16:55
13351    04:25 07 Mar
13352           19:15
13353           19:15
Name: Arrival_Time, Length: 13354, dtype: object

In [22]:
final['Time']=final['Arrival_Time'].str.split(' ').str[0]

In [23]:
final

Airline    Source Destination                  Route  \
0                 IndiGo  Banglore   New Delhi              BLR → DEL   
1              Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2            Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK   
3                 IndiGo   Kolkata    Banglore        CCU → NAG → BLR   
4                 IndiGo  Banglore   New Delhi        BLR → NAG → DEL   
...                  ...       ...         ...                    ...   
13349          Air India   Kolkata    Banglore        CCU → DEL → BLR   
13350             IndiGo   Kolkata    Banglore              CCU → BLR   
13351        Jet Airways     Delhi      Cochin        DEL → BOM → COK   
13352          Air India     Delhi      Cochin        DEL → BOM → COK   
13353  Multiple carriers     Delhi      Cochin        DEL → BOM → COK   

      Dep_Time  Arrival_Time Duration Total_Stops    Price Date Month  Year  \
0        22:20  01:10 22 Mar   2h 50m    non-stop   3897.0   24    03  2019   
1        05:50         13:15   7h 25m     2 stops   7662.0    1    05  2019   
2        09:25  04:25 10 Jun      19h     2 stops  13882.0    9    06  2019   
3        18:05         23:30   5h 25m      1 stop   6218.0   12    05  2019   
4        16:50         21:35   4h 45m      1 stop  13302.0   01    03  2019   
...        ...           ...      ...         ...      ...  ...   ...   ...   
13349    20:30  20:25 07 Jun  23h 55m      1 stop      NaN    6    06  2019   
13350    14:20         16:55   2h 35m    non-stop      NaN   27    03  2019   
13351    21:50  04:25 07 Mar   6h 35m      1 stop      NaN    6    03  2019   
13352    04:00         19:15  15h 15m      1 stop      NaN    6    03  2019   
13353    04:55         19:15  14h 20m      1 stop      NaN   15    06  2019   

        Time  
0      01:10  
1      13:15  
2      04:25  
3      23:30  
4      21:35  
...      ...  
13349  20:25  
13350  16:55  
13351  04:25  
13352  19:15  
13353  19:15  

[13354 rows x 13 columns]

In [24]:
final.drop('Arrival_Time',axis=1,inplace=True)

In [25]:
final.head()

Airline    Source Destination                  Route Dep_Time Duration  \
0       IndiGo  Banglore   New Delhi              BLR → DEL    22:20   2h 50m   
1    Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR    05:50   7h 25m   
2  Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK    09:25      19h   
3       IndiGo   Kolkata    Banglore        CCU → NAG → BLR    18:05   5h 25m   
4       IndiGo  Banglore   New Delhi        BLR → NAG → DEL    16:50   4h 45m   

  Total_Stops    Price Date Month  Year   Time  
0    non-stop   3897.0   24    03  2019  01:10  
1     2 stops   7662.0    1    05  2019  13:15  
2     2 stops  13882.0    9    06  2019  04:25  
3      1 stop   6218.0   12    05  2019  23:30  
4      1 stop  13302.0   01    03  2019  21:35

In [26]:
final.isnull().sum()

Airline           0
Source            0
Destination       0
Route             1
Dep_Time          0
Duration          0
Total_Stops       1
Price          2671
Date              0
Month             0
Year              0
Time              0
dtype: int64

In [27]:
final['Price'].isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
13349     True
13350     True
13351     True
13352     True
13353     True
Name: Price, Length: 13354, dtype: bool

In [28]:
final['Price'].mode()[0]

np.float64(10262.0)

In [29]:
final['Price']=final['Price'].fillna(final['Price'].mode()[0])

In [30]:
final['Price'].isnull().sum()

np.int64(0)

In [31]:
final

Airline    Source Destination                  Route  \
0                 IndiGo  Banglore   New Delhi              BLR → DEL   
1              Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2            Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK   
3                 IndiGo   Kolkata    Banglore        CCU → NAG → BLR   
4                 IndiGo  Banglore   New Delhi        BLR → NAG → DEL   
...                  ...       ...         ...                    ...   
13349          Air India   Kolkata    Banglore        CCU → DEL → BLR   
13350             IndiGo   Kolkata    Banglore              CCU → BLR   
13351        Jet Airways     Delhi      Cochin        DEL → BOM → COK   
13352          Air India     Delhi      Cochin        DEL → BOM → COK   
13353  Multiple carriers     Delhi      Cochin        DEL → BOM → COK   

      Dep_Time Duration Total_Stops    Price Date Month  Year   Time  
0        22:20   2h 50m    non-stop   3897.0   24    03  2019  01:10  
1        05:50   7h 25m     2 stops   7662.0    1    05  2019  13:15  
2        09:25      19h     2 stops  13882.0    9    06  2019  04:25  
3        18:05   5h 25m      1 stop   6218.0   12    05  2019  23:30  
4        16:50   4h 45m      1 stop  13302.0   01    03  2019  21:35  
...        ...      ...         ...      ...  ...   ...   ...    ...  
13349    20:30  23h 55m      1 stop  10262.0    6    06  2019  20:25  
13350    14:20   2h 35m    non-stop  10262.0   27    03  2019  16:55  
13351    21:50   6h 35m      1 stop  10262.0    6    03  2019  04:25  
13352    04:00  15h 15m      1 stop  10262.0    6    03  2019  19:15  
13353    04:55  14h 20m      1 stop  10262.0   15    06  2019  19:15  

[13354 rows x 12 columns]

In [32]:
final['Total_Stops']=final['Total_Stops'].fillna(final['Total_Stops'].mode()[0])

In [33]:
final['Route']=final['Route'].fillna(final['Route'].mode()[0])



In [36]:
final.drop('Route',axis=1,inplace=True)

In [37]:
final

Airline    Source Destination Dep_Time Duration Total_Stops  \
0                 IndiGo  Banglore   New Delhi    22:20   2h 50m    non-stop   
1              Air India   Kolkata    Banglore    05:50   7h 25m     2 stops   
2            Jet Airways     Delhi      Cochin    09:25      19h     2 stops   
3                 IndiGo   Kolkata    Banglore    18:05   5h 25m      1 stop   
4                 IndiGo  Banglore   New Delhi    16:50   4h 45m      1 stop   
...                  ...       ...         ...      ...      ...         ...   
13349          Air India   Kolkata    Banglore    20:30  23h 55m      1 stop   
13350             IndiGo   Kolkata    Banglore    14:20   2h 35m    non-stop   
13351        Jet Airways     Delhi      Cochin    21:50   6h 35m      1 stop   
13352          Air India     Delhi      Cochin    04:00  15h 15m      1 stop   
13353  Multiple carriers     Delhi      Cochin    04:55  14h 20m      1 stop   

         Price Date Month  Year   Time  
0       3897.0   24    03  2019  01:10  
1       7662.0    1    05  2019  13:15  
2      13882.0    9    06  2019  04:25  
3       6218.0   12    05  2019  23:30  
4      13302.0   01    03  2019  21:35  
...        ...  ...   ...   ...    ...  
13349  10262.0    6    06  2019  20:25  
13350  10262.0   27    03  2019  16:55  
13351  10262.0    6    03  2019  04:25  
13352  10262.0    6    03  2019  19:15  
13353  10262.0   15    06  2019  19:15  

[13354 rows x 11 columns]

In [38]:
final['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

In [39]:
final['Total_Stops']=final['Total_Stops'].map({'non-stop':0, '2 stops':2, '1 stop':1, '3 stops':3, '4 stops':4})

In [40]:
final

Airline    Source Destination Dep_Time Duration  Total_Stops  \
0                 IndiGo  Banglore   New Delhi    22:20   2h 50m            0   
1              Air India   Kolkata    Banglore    05:50   7h 25m            2   
2            Jet Airways     Delhi      Cochin    09:25      19h            2   
3                 IndiGo   Kolkata    Banglore    18:05   5h 25m            1   
4                 IndiGo  Banglore   New Delhi    16:50   4h 45m            1   
...                  ...       ...         ...      ...      ...          ...   
13349          Air India   Kolkata    Banglore    20:30  23h 55m            1   
13350             IndiGo   Kolkata    Banglore    14:20   2h 35m            0   
13351        Jet Airways     Delhi      Cochin    21:50   6h 35m            1   
13352          Air India     Delhi      Cochin    04:00  15h 15m            1   
13353  Multiple carriers     Delhi      Cochin    04:55  14h 20m            1   

         Price Date Month  Year   Time  
0       3897.0   24    03  2019  01:10  
1       7662.0    1    05  2019  13:15  
2      13882.0    9    06  2019  04:25  
3       6218.0   12    05  2019  23:30  
4      13302.0   01    03  2019  21:35  
...        ...  ...   ...   ...    ...  
13349  10262.0    6    06  2019  20:25  
13350  10262.0   27    03  2019  16:55  
13351  10262.0    6    03  2019  04:25  
13352  10262.0    6    03  2019  19:15  
13353  10262.0   15    06  2019  19:15  

[13354 rows x 11 columns]

In [41]:
final['Destination'].unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

In [42]:
final['Destination']=final['Destination'].map({'New Delhi':1, 'Banglore':2, 'Cochin':3, 'Kolkata':4, 'Delhi':5, 'Hyderabad':6})

In [43]:
final.head()

Airline    Source  Destination Dep_Time Duration  Total_Stops    Price  \
0       IndiGo  Banglore            1    22:20   2h 50m            0   3897.0   
1    Air India   Kolkata            2    05:50   7h 25m            2   7662.0   
2  Jet Airways     Delhi            3    09:25      19h            2  13882.0   
3       IndiGo   Kolkata            2    18:05   5h 25m            1   6218.0   
4       IndiGo  Banglore            1    16:50   4h 45m            1  13302.0   

  Date Month  Year   Time  
0   24    03  2019  01:10  
1    1    05  2019  13:15  
2    9    06  2019  04:25  
3   12    05  2019  23:30  
4   01    03  2019  21:35

In [44]:
final['Source'].unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

In [45]:
final['Source']=final['Source'].map({ 'Banglore':2,  'Kolkata':4, 'Delhi':5, 'Chennai':7,'Mumbai':8})

In [46]:
final.head()

Airline  Source  Destination Dep_Time Duration  Total_Stops    Price  \
0       IndiGo       2            1    22:20   2h 50m            0   3897.0   
1    Air India       4            2    05:50   7h 25m            2   7662.0   
2  Jet Airways       5            3    09:25      19h            2  13882.0   
3       IndiGo       4            2    18:05   5h 25m            1   6218.0   
4       IndiGo       2            1    16:50   4h 45m            1  13302.0   

  Date Month  Year   Time  
0   24    03  2019  01:10  
1    1    05  2019  13:15  
2    9    06  2019  04:25  
3   12    05  2019  23:30  
4   01    03  2019  21:35

In [48]:
final['Airline'].value_counts()

Airline
Jet Airways                          4746
IndiGo                               2564
Air India                            2192
Multiple carriers                    1543
SpiceJet                             1026
Vistara                               608
Air Asia                              405
GoAir                                 240
Multiple carriers Premium economy      16
Jet Airways Business                    8
Vistara Premium economy                 5
Trujet                                  1
Name: count, dtype: int64

In [54]:
from sklearn.preprocessing import LabelEncoder
labelencoder=LabelEncoder()

In [55]:
final['Airline']=labelencoder.fit_transform(final['Airline'])

In [56]:
final.head()

Airline  Source  Destination Dep_Time Duration  Total_Stops    Price Date  \
0        3       2            1    22:20   2h 50m            0   3897.0   24   
1        1       4            2    05:50   7h 25m            2   7662.0    1   
2        4       5            3    09:25      19h            2  13882.0    9   
3        3       4            2    18:05   5h 25m            1   6218.0   12   
4        3       2            1    16:50   4h 45m            1  13302.0   01   

  Month  Year   Time  
0    03  2019  01:10  
1    05  2019  13:15  
2    06  2019  04:25  
3    05  2019  23:30  
4    03  2019  21:35

In [62]:
final['Airline'].value_counts()

Airline
4     4746
3     2564
1     2192
6     1543
8     1026
10     608
0      405
2      240
7       16
5        8
11       5
9        1
Name: count, dtype: int64

In [65]:
labelencoder.classes_

array(['Air Asia', 'Air India', 'GoAir', 'IndiGo', 'Jet Airways',
       'Jet Airways Business', 'Multiple carriers',
       'Multiple carriers Premium economy', 'SpiceJet', 'Trujet',
       'Vistara', 'Vistara Premium economy'], dtype=object)


IndiGo                               2564
Air India                            2192
Multiple carriers                    1543
SpiceJet                             1026
Vistara                               608
Air Asia                              405
GoAir                                 240
Multiple carriers Premium economy      16
Jet Airways Business                    8
Vistara Premium economy                 5
Trujet                                  1